### ISSUE TITLE
"Raise network fee" assumes 1kB transaction & selects excessive fee

### ISSUE THREAD

**Issue by mbevand** [bitcoin-wallet/bitcoin-wallet#381]
When using the "raise network fee" option, Bitcoin Wallet for Android 5.14 disregards the fee category the transaction was initially sent at (ECONOMIC, NORMAL, PRIORITY) but blindly takes the PRIORITY fee value, and multiply it by 2 (hence implicitly assuming a transaction of 1kB since PRIORITY is supposed to be a value in sat/kB):

~~~
// RaiseFeeDialogFragment.java
feeRaise = data.get(FeeCategory.PRIORITY).multiply(2);
~~~

This causes unnecessarily high fees. I noticed this after I had a small 225-byte transaction stuck unconfirmed after 10 mined block despite a fee of ~28000 sat (~126 sat/B, a NORMAL fee). When I tried to raise its network fee it generated a second 192-byte tx with a fee of 320000 sat (2 multiplied by PRIORITY which is 160000 in https://wallet.schildbach.de/fees at the moment).

So in total I paid ~348000 sat for 417 bytes or ~835 sat/B, which is **4 times higher than necessary** according to https://bitcoinfees.21.co which recommends ~220 sat/B for an estimated delay of 0 blocks. 

IMHO the code should be changed to properly compute the fee in sat/B instead of assuming a 1kB transaction. And I question the validity of picking 2*PRIORITY. Shouldn't the logic of raising the fee be "pick the next higher-category fee" and only if at PRIORITY already, multiply by 2 (or 1.5 or some other less drastic value)?

**Comment by schildbach** [comment]
Raise fee is mainly meant for incoming payments, rather than outgoing. For outgoing payments, better use Priority fee to begin with if your payment is urgent.

For incoming payments, Bitcoin Wallet doesn't know how much fee the sender used. This is due to a limitation of the Bitcoin protocol. So the app must assume its very little, so basically it must pay priority fee for two transactions. I'm sure this could be optimized, PR welcome.

About the "per KB", you're right, we should change this. I'll have a look at it.

**Comment by mbevand** [comment]
Re: incoming vs outgoing: I see... Ok then fixing the "per kB" computation is sufficient for now.

**Comment by wernight** [comment]
On that same note the [`wallet/README.md`](https://github.com/bitcoin-wallet/bitcoin-wallet/blob/master/wallet/README.md) shows where one can change the fees, but there is not unit in [`wallet/assets/fees.txt`](https://github.com/bitcoin-wallet/bitcoin-wallet/blob/master/wallet/assets/fees.txt).

From the code it does:

    sendRequest.feePerKb = fees.get(feeCategory);

So it looks like they are Satoshis per KB.

**Comment by schildbach** [comment]
Fixed by #382.

@wernight Yes, the fee in fees.txt is sat per kB.
